src/controllers/dlgprefcontroller.cpp

```diff
@@ -388,7 +388,7 @@ void DlgPrefController::slotApply() {
     applyPresetChanges();
 
     // If no changes were made, do nothing
-    if (!isDirty() && (!m_pPreset || m_pPreset->isDirty())) {
+    if (!(isDirty() || (m_pPreset && m_pPreset->isDirty()))) {
```

Comment:
Why do both DlgPrefController and ControllerPreset have dirty states?

Reply:
The controller preset might be dirty because one of it's XML attributes or mappings might have changed.

DlgPrefController can be dirty when the enabled button was checked/unchecked or a different controller preset has been selected.

Reply:
`m_bDirty` is never set to true in DlgPrefController. Please remove it.

Reply:
it is set via `setDirty()` on checkbox changes and when the mapping combobox is used:
https://github.com/mixxxdj/mixxx/blob/420ceb91b45fcfedb0e4c7629362ec43677b033c/src/controllers/dlgprefcontroller.cpp#L63
https://github.com/mixxxdj/mixxx/blob/420ceb91b45fcfedb0e4c7629362ec43677b033c/src/controllers/dlgprefcontroller.cpp#L432

Reply:
Oh it's set indirectly with the setDirty function. But that is only used when showing the mapping wizard, which IIUC changes the preset and marks it dirty anyway, and when the selected mapping in the combobox has changed. Does DlgPrefController even need to track that state with a bool member? How about explicitly comparing the loaded preset to the combobox selected preset in slotApply instead of complicating it with this "dirty" state?